### PR TITLE
Update Strings used in site icon in customizer.

### PR DESCRIPTION
**Pull request by aaronjorbin**
Updates both alt text and the description string to match the general settings page.

Trac ticket: https://core.trac.wordpress.org/ticket/60641

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by aaronjorbin** [comment]
@westonruter (because customizer) @swissspidy (because internationalization) I'm wondering if either of you has thoughts on this, specifically the use of the `wp.i18n` functions inside of the templates. As these are not actually JS files, I am unsure if this will actually work as expected. I couldn't find any other uses of these functions in `src/wp-includes/customize` which is why I'm second-guessing my approach.

**Comment by swissspidy** [comment]
Yeah unfortunately I don‘t think that works, because the JS string extraction only works on JavaScript files, not JS templates inside a PHP file. That meand it couldn‘t be actually translated.

So either keep the existing approach or, if we want to use the JS i18n functions, use them in a JS file and inject into the template as just another variable. For the latter it could be a BC brack if plugins use these templates as well.

Hope that makes sense

File: src/wp-includes/customize/class-wp-customize-site-icon-control.php [hunk 1]
```diff
@@ -66,11 +66,39 @@ public function content_template() {
...
+											wp.i18n.__( 'Browser icon preview: Current image: %s' ),
+											data.attachment.alt
+										) :
+										wp.i18n.sprintf(
+											/* translators: %s: The selected image filename. */
+											wp.i18n.__(
+												'Browser icon preview: The current image has no alternative text. The file name is: %s'
+											),
+											data.attachment.filename
+										)
```
Comment (westonruter) [inline src/wp-includes/customize/class-wp-customize-site-icon-control.php]:
```suggestion
										wp.i18n.sprintf(
											<?php
											/* translators: %s: The selected image alt text. */
											echo wp_json_encode( __( 'Browser icon preview: Current image: %s' ) );
											?>,
											data.attachment.alt
										) :
										wp.i18n.sprintf(
											<?php
											/* translators: %s: The selected image filename. */
											echo wp_json_encode( __( 'Browser icon preview: The current image has no alternative text. The file name is: %s' ) );
											?>,
											data.attachment.filename
										)
```

File: src/wp-includes/customize/class-wp-customize-site-icon-control.php [hunk 2]
```diff
@@ -66,11 +66,39 @@ public function content_template() {
...
+									wp.i18n.__( 'App icon preview: Current image: %s' ),
+									data.attachment.alt
+								) :
+								wp.i18n.sprintf(
+									/* translators: %s: The selected image filename. */
+									wp.i18n.__(
+										'App icon preview: The current image has no alternative text. The file name is: %s'
+									),
+									data.attachment.filename
+								)
```
Comment (westonruter) [inline src/wp-includes/customize/class-wp-customize-site-icon-control.php]:
```suggestion
								wp.i18n.sprintf(
									<?php
									/* translators: %s: The selected image alt text. */
									echo wp_json_encode( __( 'App icon preview: Current image: %s' ) )
									?>,
									data.attachment.alt
								) :
								wp.i18n.sprintf(
									<?php
									/* translators: %s: The selected image filename. */
									echo wp_json_encode( __( 'App icon preview: The current image has no alternative text. The file name is: %s' ) );
									?>,
									data.attachment.filename
								)
```

**Review by westonruter — COMMENTED** [review]
@swissspidy How about just printing the localized strings in PHP like this?

File: src/wp-includes/customize/class-wp-customize-site-icon-control.php (hunk 1 above)
Comment (westonruter) [reply src/wp-includes/customize/class-wp-customize-site-icon-control.php]: (Updated suggestion after I apparently failed to paste it properly)

**Comment by aaronjorbin** [comment]
I like this approach. If it is also problematic, I think my next method to explore would be putting it in `src/js/_enqueues/wp/customize/controls.js`

**Comment by swissspidy** [comment]
That should work 👍

**Comment by lenaseo22** [comment]
It seems like we've hit a roadblock with the current approach, as the JavaScript string extraction only operates on JavaScript files, not JavaScript templates embedded within PHP files. This limitation means that the translation process might not work as intended.

To overcome this obstacle, we have a couple of options. We can stick with the existing approach, or if we're determined to use JavaScript internationalization functions, we could incorporate them into a separate JavaScript file and then inject them into the template as just another variable. However, this latter solution might require backward compatibility adjustments if plugins also utilize these templates.
{{{
#!html
<a href="https://bytevarsity.ca/"></a><a href="http://ukvibesonline.co.uk/"></a><a href="https://dcanews.ca/"></a><a href="https://canadahomesnews.ca/"></a><a href="https://blesa.ca/"></a><a href="https://vixrim.ca/"></a><a href="https://totimescanews.ca/"></a><a href="http://canadianews.ca/"></a><a href="http://newsag.ca/"></a><a href="http://ottawas.ca/"></a><a href="http://canadavista.ca/"></a><a href="https://cryptowavezen.com/"></a><a href="https://bitcryptomining.net/"></a><a href="https://cryptocoinsupdates.com/"></a><a href="https://forexcryptonews.net/"></a><a href="https://bitforexvista.com/"></a><a href="https://stocktradalert.com/"></a><a href="https://tradingsentiment.org/"></a><a href="https://forexinvester.com/"></a><a href="https://stocksnotify.com/"></a><a href="https://stocktrendsalert.com/"></a><a href="https://topforexebroker.com/"></a><a href="https://forexah.com/"></a><a href="https://forexinsides.com/"></a><a href="https://stocksrush.com/"></a>
}}}

I hope this clarifies the situation and provides a clearer path forward. If you have any further questions or concerns, please don't hesitate to ask.

**Pull request closed without merging**